servers/rendering/renderer_canvas_cull.cpp

```diff
@@ -435,10 +442,14 @@ void RendererCanvasCull::canvas_set_item_mirroring(RID p_canvas, RID p_item, con
 }
 
 void RendererCanvasCull::canvas_set_item_repeat(RID p_item, const Point2 &p_repeat_size, int p_repeat_times) {
+	ERR_FAIL_COND(p_repeat_size.x < 0 || p_repeat_size.y < 0);
+	ERR_FAIL_COND(p_repeat_times < 0);
```

Comment:
While disallowing negative `p_repeat_times` is kinda obvious, I'm not sure about `p_repeat_size`. Should I remove the first check? :thinking:

Reply:
Would it even work with a negative repeat size? I don't think its set up to work backwards, but it's been awhile since I looked.

Reply:
Parallax2D prevents that, but manually using RenderingServer it was possible to set negative repeat size. I think it might somehow work, but I'd suspect rect/clipping calculations doesn't account for that so likely not working correctly anyway.

Reply:
I've end up removing that `ERR_FAIL_COND(p_repeat_size.x < 0 || p_repeat_size.y < 0);` check (and not adding similar check in `RendererCanvasCull::canvas_set_item_mirroring`).

With the current code it should work fine for a negative repeat size, would simply result in adding repeats in the opposite direction first (rect would still be calculated correctly).

So e.g. for `p_repeat_times = 1`:
- `p_repeat_size = Vector2(64, 0)` would add a single repeat to the right (+X),
- `p_repeat_size = Vector2(-64, 0)` would add a single repeat to the left (-X),
- `p_repeat_size = Vector2(0, 64)` would add a single repeat to the bottom (+Y),
- `p_repeat_size = Vector2(0, -64)` would add a single repeat to the top (-Y).

For even `p_repeat_times` results should be the same regardless of the size's signs (same repeat count per right/left and top/bottom).

Note that both Parallax2D/ParallaxLayer do prevent negative repeat sizes themselves, nothing changes about it. But since negative size works in a sensible manner for the current implementation I see no point in preventing such usage when manually using the RenderingServer.